src/rdkafka_queue.c

```diff
@@ -687,7 +687,20 @@ rd_kafka_queue_t *rd_kafka_queue_get_consumer (rd_kafka_t *rk) {
 	return rd_kafka_queue_new0(rk, rk->rk_cgrp->rkcg_q);
 }
 
+rd_kafka_queue_t *rd_kafka_queue_get_partition (rd_kafka_t *rk,
+                                                const char* topic,
+                                                int32_t partition) {
+    shptr_rd_kafka_toppar_t *s_rktp = rd_kafka_toppar_get2(rk, topic, partition,
+                                                           0, 1);
+    rd_kafka_toppar_t *rktp = rd_kafka_toppar_s2i(s_rktp);
+    rd_kafka_queue_t *result = rd_kafka_queue_new0(rk, rktp->rktp_fetchq);
+    rd_kafka_toppar_destroy(s_rktp);
+
```

Comment:
I changed this to return NULL if s_rktp isn't valid. This makes matches the behavior of rd_kafka_queue_get_consumer. Otherwise, I'd have to change rd_kafka_queue_get_partition to have a drastically different interface than other functions of similar functionality.